### PR TITLE
fix(BoundingBox): add more primitive intersections

### PR DESCRIPTION
**Pull request by BotellaA**

### CODE REVIEW

File: include/geode/geometry/bounding_box.h [hunk 1]
```diff
@@ -61,6 +66,18 @@ namespace geode
 
         bool intersects( const InfiniteLine< dimension >& line ) const;
 
+        template < index_t T = dimension >
+        typename std::enable_if< T == 2, bool >::type intersects(
+            const Segment< T >& segment ) const;
+
+        template < index_t T = dimension >
+        typename std::enable_if< T == 2 || T == 3, bool >::type intersects(
```
Comment (panquez) [inline include/geode/geometry/bounding_box.h:74]: If i have rightly understood impl, it intersects (at least in 2D) if and only if the edges (or the facets of tetra) intersect the bbox. What is the aimed behavior when the bbox is included in the triangle?


At least, these information should be clarify in documentation.

File: src/geode/geometry/bounding_box.cpp [hunk 2]
```diff
@@ -128,6 +133,59 @@ namespace geode
...
+            const auto& vertices = tetra.vertices();
+            return intersects( { vertices[0].get(), vertices[1].get(),
+                       vertices[2].get() } )
+                   || intersects( { vertices[0].get(), vertices[1].get(),
+                       vertices[3].get() } )
+                   || intersects( { vertices[0].get(), vertices[2].get(),
+                       vertices[3].get() } )
+                   || intersects( { vertices[1].get(), vertices[2].get(),
+                       vertices[3].get() } );
+        }
```
Comment (panquez) [inline src/geode/geometry/bounding_box.cpp:153]: || bbox.center() is in the tetrahedron

File: include/geode/geometry/bounding_box.h (hunk 1 above)
Comment (BotellaA) [reply include/geode/geometry/bounding_box.h:74]: The aimed behavior is intersection if the element is crossing, is inside, or is containing the bbox. (I am adding this in the doc)

The implementation does not really detect intersections, it uses the separating axis method which detects overlapping intervals of the projected edge (or facet) and the bbox on several axes. It matches the expected behavior. Tests include this special case.

File: src/geode/geometry/bounding_box.cpp (hunk 2 above)
Comment (BotellaA) [reply src/geode/geometry/bounding_box.cpp:153]: No need. The intersections tests are enough

**Review by panquez — APPROVED** [review]

**Pull request merged**